### PR TITLE
Add oracle example

### PR DESCRIPTION
**Pull request by Lbqds**

### CODE REVIEW

**Comment by Lbqds** [comment]
The CI will pass after we update web3-react and extension-wallet to the latest SDK.

File: contracts/dex/token_pair.ral [hunk 1]
```diff
@@ -55,7 +55,7 @@ Contract TokenPair(
 
   @using(updateFields = true)
   fn update(newReserve0: U256, newReserve1: U256) -> () {
-    let max = 1 << 112 - 1
+    let max = (1 << 112) - 1
```
[the remark this answers is not in the document]
Comment (polarker) [reply contracts/dex/token_pair.ral]: Let's use bracket everywhere. Operator precedence is hard to remember 😅

[the remark this answers is not in the document]
Comment (polarker) [reply contracts/dex/token_pair.ral]: I will redeploy everything later with large amount of token

**Review by polarker — APPROVED** [review]
LGTM, just one remark.

**Pull request merged**